osdep/macOS_mpv_helper.swift

```diff
@@ -21,15 +21,26 @@ import OpenGL.GL3
 
 let glDummy: @convention(c) () -> Void = {}
 
+extension Bool {
+    init(_ num: Int32) {
+        self.init(num > 0)
+    }
+}
+
 class MPVHelper: NSObject {
 
     var mpvHandle: OpaquePointer?
     var mpvRenderContext: OpaquePointer?
     var mpvLog: OpaquePointer?
     var inputContext: OpaquePointer?
     var mpctx: UnsafeMutablePointer<MPContext>?
+    var macOptsPtr: UnsafeMutablePointer<macos_opts>?
     var fbo: GLint = 1
 
+    var vo: UnsafeMutablePointer<vo> { get { return mpctx!.pointee.video_out } }
+    var opts: mp_vo_opts { get { return mpctx!.pointee.opts.pointee.vo.pointee } }
```

Comment:
Any internal MPContext access is straight out. I'd just randomly break the OSX parts when modfying them. There's a reason literally everything in the codebase goes out of its way to not access MPContext.

Reply:
then i am already doing something wrong because i access the `vo` struct inside the MPContext for the `vo_calc_window_geometry2` function. i don't really fancy recreating that function because i have no access to the vo struct.